src/lib/com/yetanalytics/xapipe/filter.clj

```diff
@@ -89,23 +91,20 @@
                        (map :id (filter #(= (:type %) "ActivityType") concepts)))
         att-ids    (or (not-empty attachment-usage-types)
                        (map :id (filter #(= (:type %) "AttachmentUsageType") concepts)))
-        validators (cond-> []
-                     (empty? concept-types)
-                     (concat (concept/verb-validators vrb-ids)
-                             (concept/activity-type-validators act-ids)
-                             (concept/attachment-usage-validators att-ids))
-                     (some #{"Verb"} (set concept-types))
-                     (into (concept/verb-validators vrb-ids))
-                     (some #{"ActivityType"} (set concept-types))
-                     (into (concept/activity-type-validators act-ids))
-                     (some #{"AttachmentUsageType"} (set concept-types))
-                     (into (concept/attachment-usage-validators att-ids)))]
+        templates (cond-> []
+                    (empty? concept-types)
+                    (concat (concept/verb-templates vrb-ids)
+                            (concept/activity-type-templates act-ids)
+                            (concept/attachment-usage-templates att-ids))
+                    (some #{"Verb"} (set concept-types))
+                    (into (concept/verb-templates vrb-ids))
+                    (some #{"ActivityType"} (set concept-types))
+                    (into (concept/activity-type-templates act-ids))
+                    (some #{"AttachmentUsageType"} (set concept-types))
+                    (into (concept/attachment-usage-templates att-ids)))
+        validators (per/compile-templates->validators templates :validate-templates? false)]
```

Comment:
whats that key-arg do?

Reply:
I had a realization looking through pers. is this on account of the ID clash?

Reply:
Yep! If you don't add it you get an (appropriate) error for non-unique template ids

Reply:
i could happily add some random to the id gen, if that's a nicer solution. I guess honestly validating a mock is kind of nonsense anyway so maybe not

Reply:
Yeah I'd advise leaving it